src/helpers.js

```diff
@@ -31,71 +31,97 @@ function exec(command: string, parameters: Array<string>): Promise<{ stdout: str
   })
 }
 
-export function apmInstall(dependencies: Array<Dependency>, progressCallback: ((packageName: string, status: boolean) => void)): Promise<Map<string, Error>> {
+export function apmInstall(
+  dependencies: Array<Dependency>,
+  progressCallback: (packageName: string, status: boolean) => void,
+): Promise<Map<string, Error>> {
   const errors = new Map()
-  return Promise.all(dependencies.map(function(dep) {
-    return exec(atom.packages.getApmPath(), ['install', dep.version ? `${dep.url}@${dep.version}` : dep.url, '--production', '--color', 'false'])
-      .then(function(output) {
-        const successful = VALIDATION_REGEXP.test(output.stdout) && VALID_TICKS.has(VALIDATION_REGEXP.exec(output.stdout)[2])
-        progressCallback(dep.name, successful)
-        if (!successful) {
-          const error = new Error(`Error installing dependency: ${dep.name}`)
-          error.stack = output.stderr
-          throw error
-        }
-      })
-      .catch(function(error) {
-        errors.set(dep.name, error)
-      })
-  })).then(function() {
+  return Promise.all(
+    dependencies.map(function(dep) {
+      return exec(atom.packages.getApmPath(), ['install', dep.url || dep.name, '--production', '--color', 'false'])
+        .then(function(output) {
+          let successful = VALIDATION_REGEXP.test(output.stdout)
+          if (successful) {
+            const match = VALIDATION_REGEXP.exec(output.stdout)
+            successful = match && VALID_TICKS.has(match[2])
+          }
+          progressCallback(dep.name, !!successful)
+          if (!successful) {
+            const error = new Error(`Error installing dependency: ${dep.name}`)
+            error.stack = output.stderr
+            throw error
+          }
+        })
+        .catch(function(error) {
+          errors.set(dep.name, error)
+        })
+    }),
+  ).then(function() {
     return errors
   })
 }
 
-const DEPENDENCY_REGEX = /^([^#:]+)(?:#([^:]+))?(?::(.+))?$/
+const DEPENDENCY_REGEX_VERSION = /(.*?):.*/
+const DEPENDENCY_REGEX_GIT = /(.*?)#.*/
 export async function getDependencies(packageName: string): Promise<Array<Dependency>> {
-  const toReturn = []
   const packageModule = atom.packages.getLoadedPackage(packageName)
   const packageDependencies = packageModule && packageModule.metadata['package-deps']
 
-  if (packageDependencies) {
-    for (const entry of (packageDependencies: Array<string>)) {
-      const matches = DEPENDENCY_REGEX.exec(entry)
-      if (matches === null) {
-        console.error('[Package-Deps] Error parsing dependency of', packageName, 'with value:', entry)
-        continue
+  if (!Array.isArray(packageDependencies)) {
+    console.error(`[Package-Deps] Unable to get loaded package '${packageName}'`)
+    return []
+  }
+  return (await Promise.all(
+    packageDependencies.map(async function(entry) {
+      let url = null
+      let name = entry
+      let version = null
+
+      if (DEPENDENCY_REGEX_VERSION.test(entry)) {
```

Comment:
If you are going to use the results anyway, why not just check the return value of `exec`?